From 96e67dc3b493a96a8ad880a7ebbb44012f9346cf Mon Sep 17 00:00:00 2001
From: Ted Charles <me@tedcharles.net>
Date: Sat, 15 Aug 2026 13:35:42 -0400
Subject: [PATCH] Add Emscripten WebSocket transport for RTCW MP

POSIX sockets in the browser never reach the dedicated arena. Link
websocket.js so connect 127.0.0.1:27960 is carried over the managed /ws
UDP proxy.
---
 MP/Makefile | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/MP/Makefile b/MP/Makefile
index 00166ba..a6addb0 100644
--- a/MP/Makefile
+++ b/MP/Makefile
@@ -1074,6 +1074,8 @@ ifeq ($(PLATFORM),emscripten)
     -sEXPORTED_FUNCTIONS=['_main','_malloc','_free'] \
     -sEXPORTED_RUNTIME_METHODS=['callMain','FS','IDBFS','stringToNewUTF8'] \
     -lidbfs.js \
+    -lwebsocket.js \
+    -sSOCKET_DEBUG=0 \
     -sINVOKE_RUN=0 \
     -sEXIT_RUNTIME=0 \
     -sENVIRONMENT=web
-- 
2.53.0

